app.js

```diff
@@ -28,6 +33,8 @@ app.use((err, req, res, next) => {
 app.use((err, req, res, next) => {
   if (err.code === "22P02") {
     res.status(400).send({ msg: "Bad request" });
+  } else if (err.code === "23503") {
+    res.status(404).send({ msg: "Article/User not found!" });
```

Comment:
See my comment in the test file - better to have a general message here than try to cover all bases